### PR TITLE
(fix) Allow cancelling orders and address various small fixes

### PR DESCRIPTION
**Pull request by ibacher**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR does various minor fixes to the order details table:

* Orders are now correctly cancelled, as cancelling an order now generates a DISCONTINUE order
* Stricter typings
* Moved a bunch of style-determining code into CSS
* Used core translations in appropriate spots
* Added the ability to translate most untranslatable text in the component
* Switch to using framework-provided icons
* Better formatting for status and urgency information (NO SHOUTING UNLESS ITS URGENT!)
* Removed a bunch of unused code (mostly sorting)

## Screenshots

Screenshot of updated orders table

### CODE REVIEW

**Comment by ibacher** [comment]
Seems this introduced a new bug, which I need to look into...

File: packages/esm-patient-orders-app/src/components/order-details-table.scss [hunk 1]
```diff
@@ -1,7 +1,8 @@
 @use '@carbon/styles/scss/colors';
 @use '@carbon/styles/scss/spacing';
+@use '@carbon/styles/scss/theme';
```
Comment (denniskigen) [inline packages/esm-patient-orders-app/src/components/order-details-table.scss:3]: Doesn't seem like we're using this import.

File: packages/esm-patient-orders-app/src/components/order-details-table.scss [hunk 2]
```diff
@@ -30,19 +31,66 @@
...
+
+  &[data-status='received'] {
+    background-color: colors.$blue-20;
+  }
+
+  &[data-status='in progress'] {
+    background-color: colors.$cyan-20;
+  }
+
+  &[data-status='on-hold'] {
```
Comment (denniskigen) [inline packages/esm-patient-orders-app/src/components/order-details-table.scss:69]:
```suggestion
  &[data-status='on-hold'] {
```

Should this be `&[data-status='on hold'] {` instead?

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx [hunk 3]
```diff
@@ -1,6 +1,5 @@
-import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
-import capitalize from 'lodash-es/capitalize';
-import orderBy from 'lodash-es/orderBy';
+import React, { type ReactElement, type ReactNode, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { capitalize, lowerCase } from 'lodash-es';
 import { useTranslation } from 'react-i18next';
 import { useReactToPrint } from 'react-to-print';
 import {
```
Comment (denniskigen) [inline packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:5]: Looks like we can remove the `Tag` component import as we're no longer using it.

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx [hunk 4]
```diff
@@ -328,7 +329,7 @@ const OrderDetailsTable: React.FC<OrderDetailsProps> = ({ title, patientUuid, sh
               {showAddButton && (
                 <Button
                   kind="ghost"
-                  renderIcon={(props) => <Add size={16} {...props} />}
+                  renderIcon={() => <AddIcon size={16} />}
```
Comment (denniskigen) [inline packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:332]: Looks like there's a regression with the icon buttons:

CleanShot 2024-07-08 at 10  27 05@2x

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx [hunk 5]
```diff
@@ -301,7 +297,12 @@ const OrderDetailsTable: React.FC<OrderDetailsProps> = ({ title, patientUuid, sh
           displayText={
             selectedOrderTypeUuid === null
               ? t('orders', 'Orders')
-              : (orderTypes?.find((x) => x.uuid === selectedOrderTypeUuid)).display + 's'
+              : // t('Drug Order_few', 'Drug Orders')
+                // t('Test Order_few', 'Test Orders')
+                t(`${(orderTypes?.find((x) => x.uuid === selectedOrderTypeUuid)).name}`, {
+                  count: 3,
+                  default: (orderTypes?.find((x) => x.uuid === selectedOrderTypeUuid)).name,
```
Comment (denniskigen) [inline packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:304]: Maybe extract `orderTypes?.find((x) => x.uuid === selectedOrderTypeUuid)).name` into a variable?

**Review by denniskigen — COMMENTED** [review]
Mostly LGTM. I've left some minor suggestions.

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx (hunk 4 above)
Comment (ibacher) [reply packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx]: Aha! The props are important. I'll look into the color thing, but I think that's best fixed in the framework

File: packages/esm-patient-orders-app/src/components/order-details-table.scss (hunk 2 above)
Comment (ibacher) [reply packages/esm-patient-orders-app/src/components/order-details-table.scss]: Whoops!

File: packages/esm-patient-orders-app/src/components/order-details-table.scss [hunk 6]
```diff
@@ -30,19 +30,66 @@
...
+  &[data-urgency='routine'] {
+    background-color: colors.$green-20;
+  }
+
+  &[data-urgency='urgent'] {
+    background-color: colors.$red-20;
+  }
+}
+
+.statusPill {
```
Comment (pirupius) [inline packages/esm-patient-orders-app/src/components/order-details-table.scss:49]: 💯

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx [hunk 7]
```diff
@@ -317,7 +318,7 @@ const OrderDetailsTable: React.FC<OrderDetailsProps> = ({ title, patientUuid, sh
               {showPrintButton && (
                 <Button
                   kind="ghost"
-                  renderIcon={(props) => <Printer size={16} {...props} />}
+                  renderIcon={(props) => <PrinterIcon {...props} size={16} />}
```
Comment (pirupius) [inline packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:321]: Out of curiosity, is the design for iconography from the framework meant to inherit designs from the parent? for example in this case the Button stylings like color?

Comment (ibacher) [reply packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:321]: They should inherit them, but I'm still untangling the mess of CSS to get things right. In this case, Carbon defines a selector like this:

```css
.cds--btn--ghost:not([disabled]) svg {
  fill: var(--cds-icon-primary, #161616);
}
```

But also setting `fill="currentColor"` on the icon, so it effectively ignores the CSS. Originally, I was doing something similar too, but it caused issues with getting the color right in some other cases. I'm still fiddling with getting this working in all cases...

Comment (ibacher) [reply packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:321]: The `props` here are basically passing in some additional padding...

**Pull request merged**